test/strings/annotated.jl

```diff
@@ -58,6 +58,9 @@
                              (3:3, :val => 2)])),
              Base.AnnotatedString("abc", [(1:2, :val => 1),
                                   (2:3, :val => 2)]))
+    @test chopprefix(sprint(show, str), "Base.") ==
+        "AnnotatedString{String}(\"some string\", , [(1:4, :thing=>0x01), (6:11, :other=>0x02), (1:11, :all=>0x03)])"
+    @test sprint(show, str, MIME("text/plain")) == "\"some string\""
```

Comment:
```suggestion
    @test sprint(show, str, MIME("text/plain")) == "\"some string\""
    @test eval(Meta.parse(repr(str))) == str
```
Just to make sure it repr-evals.